python/services/github_service.py

```diff
@@ -65,6 +67,33 @@ def __init__(self, org_token: str, organisation_name: str) -> None:
         ), execute_timeout=60)
         self.organisation_name: str = organisation_name
 
+    def archive_inactive_repositories(self, repo_type_to_archive: str, last_active_cutoff_date: datetime,
+                                      allow_list: list[str]) -> None:
+        for repo in self.__get_repos_to_consider_for_archiving(repo_type_to_archive):
+            if self.__is_repo_ready_for_archiving(repo, last_active_cutoff_date, allow_list):
+                repo.edit(archived=True)
+
+    def __get_repos_to_consider_for_archiving(self, repository_type: str) -> list[Repository]:
+        repositories = list(
+            self.github_client_core_api.get_organization(self.organisation_name).get_repos(type=repository_type))
+        return [repository for repository in repositories if not (repository.archived or repository.fork)]
+
+    def __is_repo_ready_for_archiving(self, repository, last_active_cutoff_date, allow_list: list[str]) -> bool:
+        latest_commit_position = 0
+        commit = None
+        try:  # Try block needed as get_commits() can cause exception when a repository has no commits as GH returns negative result.
```

Comment:
My [commit](https://github.com/ministryofjustice/operations-engineering/commit/b2a200c823361dc11b092fa8936d55c34188bfb8) added that code, the [pygithub](https://github.com/PyGithub/PyGithub/blob/8c4b9465e66afebb2c8ee55673a16b43113fd8db/github/Repository.py#L1827) library says it returns a pagination list, may be an exception is raised if there is no commit on the repo for it to create the list

Reply:
Happy to have a play and see what it does now 🧪

I believe the list will be empty on return and we can just check its size instead of throwing the exception 👀